Add bluechi_is_online node module and integration test

nsimsolo opened the pull request:
Adding test for bluechi-is-online node.
'bluechi-is-online node' command is verifying on the controller machine (check and/or monitor) the connection of the given node to the controller
When the node is online, it exits with a status code 0 and does not produce any CLI output.

Fixes: https://github.com/eclipse-bluechi/bluechi/issues/982
Signed-off-by: Nisim Simsolo [nsimsolo@redhat.com](mailto:nsimsolo@redhat.com)

mwperina commented on tests/tests/tier0/bluechi-is-online-node/test_bluechi_is_online_node.py:
```diff
@@ -0,0 +1,46 @@
+def check_execs(ctrl: BluechiControllerMachine, node_name: str):
+    # Check that the BlueChi controller is online at the startup
+    result = ctrl.bluechi_is_online.node_is_online(node_name=node_name)
+    assert result, f"bluechi controller for node {node_name} should be online"
```
The message should be: "bluechi-agent for node {node_name} should be online"

nsimsolo replied:
Fixed

mwperina commented on tests/tests/tier0/bluechi-is-online-node/test_bluechi_is_online_node.py:
```diff
@@ -0,0 +1,46 @@
+
+    # Stop the BlueChi controller
+    ctrl.systemctl.stop_unit("bluechi-controller")
+    assert ctrl.wait_for_unit_state_to_be("bluechi-controller", "inactive")
```
You should stop the bluechi-agent on node-foo and not the controller itself

nsimsolo replied:
Done

mwperina commented on tests/tests/tier0/bluechi-is-online-node/test_bluechi_is_online_node.py:
```diff
@@ -0,0 +1,46 @@
+
+    # Check that the BlueChi controller is offline
+    result = ctrl.bluechi_is_online.node_is_online(node_name=node_name)
+    assert not result, f"Bluechi controller for node {node_name} should be offline"
```
The message should be "bluechi-agent for node {node_name} should be offline"

nsimsolo replied:
Fixed

mwperina reviewed (changes requested):
Can't we put the whole `.idea` directory into `.gitignore` file?

mwperina reviewed (changes requested):
Test looks good, but please remove this `.idea` directory from the commit

nsimsolo committed:
Add bluechi_is_online node module and integration test

- Update .gitignore to ignore .idea and inpectionProfiles
- Adding test for bluechi-is-online node.
  'bluechi-is-online node' command is verifying on the controller
  machine (check and/or monitor) the connection of the given node
  to the controller
  When the node is online, it exits with a status code 0
  and does not produce any CLI output.

Fixes: https://github.com/eclipse-bluechi/bluechi/issues/982
Signed-off-by: nsimsolo <nsimsolo@redhat.com>

mwperina reviewed (approved):
LGTM

mkemel reviewed (approved):
LGTM

engelmi reviewed (approved):
LGTM
